citesphere/src/main/java/edu/asu/diging/citesphere/web/validation/CitationConceptValidator.java

```diff
@@ -0,0 +1,44 @@
+package edu.asu.diging.citesphere.web.validation;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+import edu.asu.diging.citesphere.core.model.IUser;
+import edu.asu.diging.citesphere.core.service.ICitationConceptManager;
+import edu.asu.diging.citesphere.web.forms.CitationConceptForm;
+
+public class CitationConceptValidator implements Validator {
+
+    @Autowired
+    private ICitationConceptManager conceptManager;
+
+    @Override
+    public boolean supports(Class<?> paramClass) {
+        return CitationConceptForm.class.equals(paramClass);
+    }
+
+    @Override
+    public void validate(Object target, Errors errors) {
+        CitationConceptForm conceptForm = (CitationConceptForm) target;
+        IUser user = (IUser) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        if (!errors.hasErrors()) {
+            if (conceptForm.getAccessMode().equals("edit")
+                    && conceptManager.get(conceptForm.getConceptId()) != null
+                    && conceptManager.get(conceptForm.getConceptId()).getOwner() != null
+                    && !(conceptManager.get(conceptForm.getConceptId()).getOwner().getUsername()
+                            .equals(user.getUsername()))) {
+                errors.rejectValue("uri", "uri", "Only the owner can edit a Concept.");
+            } else if (conceptForm.getAccessMode().equals("edit")
+                    && !conceptManager.get(conceptForm.getConceptId()).getUri()
+                            .equals(conceptForm.getUri())
+                    && conceptManager.getByUriAndOwner(conceptForm.getUri(), user) != null) {
+                errors.rejectValue("uri", "uri", "Concept with this uri already exists!");
+            } else if (conceptForm.getAccessMode().equals("add")
+                    && conceptManager.getByUriAndOwner(conceptForm.getUri(), user) != null) {
+                errors.rejectValue("uri", "uri", "Concept with this uri already exists!");
+            }
```

Comment:
all these else ifs are pretty confusing. it'll be much more easily readable if you have two ifs (one for edit, one for add) that call submethods. and why don't you call getByUriAndOwner once and then use the resulting object (you can also compare that id with the form for the check in line 33, which would be much easier to understand).

Reply:
If you first check if concept is null and then return if it is null, you can loose a whole bunch of if-statement code and then only test if the uri is already in used or if the concept already exists.